### PR TITLE
Restructuring, updated README

### PR DESCRIPTION
**Pull request by kevinkk525**
This might just work better than before. Pulled updates first, then pushed all changes at once and implemented your changes. Should now be mergeable. Leaving the other pull request open, just in case.
I did test all files but before I implemented your new changes to make the files mergeable.

Also I came across these points during my restructuring and testing:

1. Could it be interesting to implement the option for a custom message to be send after every reconnect? And I mean directly after the reconnect after the id has been sent but before the next message in the queue gets send. This way a server application could receive some configuration that might be needed to interprete the messages correctly, e.g. to start the correct protocol/app.
2. What do you think about distribution using upip/pypi (once it's finished of course)
3. As you will see in the README, repository names with an "-" are generally bad for python as they have to be renamed before they can be used as a python package. Guess this is more like an information because changing the repository name is not possible and would require a completely new repo.
4. Added a detailed how-to for getting the package onto the microcontroller. Copying just single files won't work now. Please check if that is what you like or if it is too complicated.
5. If the esp8266 has no wifi credentials, the client goes crazy and tries to connect very often as there is no wait statement for this case. Might not be a real problem though.
6. If there's a wrong server configuration in local.py, the client will just wait forever until he can connect. Would it be better to reset the microcontroller after a certain time?
7. Sometimes I had the problem that the client did not reconnect to the server but interrupting it showed that it was just waiting. Sorry I can't give more details about that.. Might be not enough to even start looking into it.
8. In the README you say that there is the need to cross compile the client.py on a daily build. Is that because uasyncio is not frozen then? Why not freeze uasyncio then?
9. I get 19.5kB of free RAM with frozen uasyncio in daily build but not frozen micropython_iot. In the README you have quite different values. (I finally got that free RAM bug fixed in my firmware by pulling the latest updates...)

### CODE REVIEW

File: example_remote_control/README.md [hunk 1]
```diff
@@ -7,7 +7,6 @@ the LED on another.
 
  1. `s_comms_cp.py` The server application. Run under CPython 3.5+.
  2. `c_comms_tx.py` Transmitting client. Expects a switch to ground on GPIO0.
- Requires `primitives.py` from this repo and `aswitch.py` from
- [this repo](https://github.com/peterhinch/micropython-async).
+ Requires `primitives.py` and `aswitch.py` from this repo [included in this package for convenience].
```
Comment (kevinkk525) [inline example_remote_control/README.md]: I included these file (stripped down version of asyn) for convenience in trying the examples.

File: example_remote_control/local.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+MY_ID = '1\n'  # Client-unique \n terminated string. Copy to local.py on device
```
Comment (kevinkk525) [inline example_remote_control/local.py]: a local.py file for every example package.

File: README.md [hunk 3]
```diff
@@ -151,28 +149,71 @@ constants must be common to all clients and the server:
...
 
 If a daily build is used it will be necessary to
 [cross compile](https://github.com/micropython/micropython/tree/master/mpy-cross)
 `client.py`
 
-To run the demo the file `local.py` should be edited for the server IP address.
-The demo supports up to four clients. Each client's `local.py` should be edited
-to give each client a unique client ID in range 1..4. Note that the ESP8266
-must have a stored network connection to access the server.
+To get the files onto your ESP8266 do NOT copy single files as this repository is
```
Comment (kevinkk525) [inline README.md]: rshell synchronization of the package.
Sadly the image file causes a little error message but rshell rsync has no exclude option yet.

File: README.md [hunk 4]
```diff
@@ -151,28 +149,71 @@ constants must be common to all clients and the server:
...
+micropython/ports/esp8266 directory.
+
+## 3.2 Usage
+
+On the esp8266 you can now run every example using the following syntax:
+```
+from micropython_iot.examples import c_app
+from micropython_iot.example_remote_control import c_comms_tx
+from micropython_iot.example_remote_control import c_comms_rx
+from micropython_iot.qos import c_qos
```
Comment (kevinkk525) [inline README.md]: A little unusal way of starting a python file on unix for the average user but works perfectly.

File: README.md [hunk 5]
```diff
@@ -300,9 +351,10 @@ A basic server-side application has this form:
 ```python
 import asyncio
 import json
-import server_cp as server
+from micropython_iot import server_cp as server
```
Comment (kevinkk525) [inline README.md]: this imports from the renamed package "micropython_iot" as "micropython-iot" is not allowed in python.

File: comms/c_comms.py [hunk 6]
```diff
@@ -1,61 +0,0 @@
-# c_comms.py Client-side application demo
```
Comment (kevinkk525) [inline comms/c_comms.py]: Removed this file as it was exactly the same as the normal example.

**Comment by kevinkk525** [comment]
I retested all files with your new commits included and they work correctly. This pull request can be merged I think, closing the other one to prevent confusion.

**Comment by peterhinch** [comment]
Thank you, from a quick initial review it looks good. I'll merge this later today or tomorrow.

I'll properly consider the points you raised later, but this is my first take. I'm bothered by 7.: I haven't seen this behaviour in many hours of testing. Points 5 and 6 have been concerning me, but I've been avoiding changing the client while you were working on it.

FYI I've refactored the server code fairly extensively. After pulling your changes I'll do some more resilience testing. I have a test that's been running for about two weeks with ESP8266 devices running 24/7. I particularly like the fact that I can ssh into the RaspberryPi, kill the server, push a new version to it and restart it. The clients just reconnect and carry on...

**Comment by kevinkk525** [comment]
I would not worry about 7 too much. I tried so many different code versions and configurations, maybe I just remember that I had problems but not if it had something to do with a particular code version or configuration. I'm sure I'll come across the problem again if it persists.

Although I have a server code that is a lot bigger, it's good to have a minimal code that can run on any hardware and even micropython itself. My server code has a different goal being the base for an application layer and therefore the focus is more on high level fetures than on simplicity. Once I get that application layer done for client and server, I'll publish both and will be replacing my mqtt system at home with it. Then we shall see about resilience in that environment too.

**Pull request merged**